Provide optional support for emulation without `apisetschema.dll`

Hi @kgwinnup and @jholowczak!

I stumbled upon Binee today, and what a pleasure it has been to start diving into it. You've essentially managed to capture an idea I've been playing around with myself for quite some time, and made it into a beautiful working system. Thanks for sharing Binee with the open source community!

As I wanted to take `binee` out for a spin, I started with a simple "hello world" sample (see `foo.exe` and `foo.go` below).

The first issue I ran into was *file 'apisetschema.dll' not found*, which is expected, as I have not (yet) downloaded the docker image. I'm currently travelling so downloading 10 GB would be limiting.

```bash
$ binee foo.exe
2019/11/21 21:01:44 file 'apisetschema.dll' not found
```

I know the rationale for implementing support for `apisetschema.dll` as there may exist several versions of a given DLL. However, as there are quite a few PE binaries that are capable of running without `apisetschema.dll` present, it would seem preferable to also add optional support using `binee` without requiring `apisetschema.dll` to be present.

On the system I'm currently running, I have access to all DLLs used by the sample set of binaries I'd like to analyze, but I do not have access to `apisetschema.dll`.

How much effort would be require to allow `binee` to analyze PE executables without requiring `apisetschema.dll` to be present?

I may peek around a bit in the code and see if I can make this optional, or if it would require a redesign of the DLL loader.

Wish you all the best and happy coding!

Cheers,
Robin

---

Contents of `foo.go`:
```go
package main

func main() {
	println("foo")
}
```

Command used to compile `foo.go`:

```bash
GOARCH=386 GOOS=windows go build -o foo.exe foo.go
```

I tried the config file, but for some reason it doesn't seem to work. Using the dummy `os/win10_32` symlink workaround worked however.

Contents of `win.yaml `:

```bash
$ cat win.yaml 
root: "/home/u/.wine/drive_c/"
```

Using `win.yaml` config:
```bash
$ binee -c win.yaml foo_32.exe
error finding file ntdll.dll
error finding file kernel32.dll
error finding file kernel32.dll
2019/11/21 21:56:35 unable to locate DLL "ntdll.dll"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x55b090]

goroutine 1 [running]:
github.com/carbonblack/binee/windows.(*WinEmulator).createLdrEntry(0xc000148900, 0x0, 0x1, 0x1)
	/home/u/Desktop/binee/windows/loader.go:291 +0x70
...
```

Using `os/win10_32` symlink workaround:

```
$ mkdir os
$ ln -s /home/u/.wine/drive_c/ os/win10_32
$ binee foo_32.exe
panic: interface conversion: interface {} is *pefile.OptionalHeader32P, not *pefile.OptionalHeader32

goroutine 1 [running]:
github.com/carbonblack/binee/windows.(*WinEmulator).createLdrEntry(0xc000114900, 0xc000138100, 0x1, 0x9)
	/home/u/Desktop/binee/windows/loader.go:303 +0x3ec
...
```

Notice, the workaround gets further (crashes in a different place).

--- a/windows/loader.go
+++ b/windows/loader.go
@@ -377,6 +377,10 @@ func retrieveDllFromDisk(cur map[string]*pefile.PeFile, apiset *pefile.PeFile, s
 	// get realDll name on disk
 	// for apiset recurse through each real dll in the apisets list
 	if strings.Compare(name[:4], "api-") == 0 {
+		if apiset == nil {
+			fmt.Fprintf(os.Stderr, "error loading dll %s; unable to locate \"apisetschema.dll\"\n", name)
+			return
+		}
 		apiset_len := len(apiset.Apisets[name[0:len(name)-6]]) - 1
 		if apiset_len >= 0 {
 			realDll = apiset.Apisets[name[0:len(name)-6]][apiset_len]
@@ -758,10 +762,11 @@ func (emu *WinEmulator) initPe(pe *pefile.PeFile, path string, arch, mode int, a
 
 	// load Apisetschema dll for mapping to real dlls
 	apisetPath, err := util.SearchFile(emu.SearchPath, "apisetschema.dll")
-	if err != nil {
-		return err
+	var apiset *pefile.PeFile
+	if err == nil {
+		// only load apisetschema.dll if present.
+		apiset, _ = pefile.LoadPeFile(apisetPath)
 	}
-	apiset, _ := pefile.LoadPeFile(apisetPath)
 
 	// create the main map to hold all name/realdll mappings to actual PeFile object
 	peMap := make(map[string]*pefile.PeFile)

